### PR TITLE
build: ibmi support for cmake

### PR DESCRIPTION
**Pull request by ThePrez**
Added the proper OS400 clause so cmake can be used with IBM i. 
cc @libuv/ibmi for awareness

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,6 +152,21 @@ if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
...
+       _XOPEN_SOURCE=500)
+  list(APPEND uv_sources
+    src/unix/aix-common.c
+    src/unix/ibmi.c
+    src/unix/posix-poll.c
+    src/unix/no-fsevents.c
+    src/unix/no-proctitle.c
+    )
+endif()
+
```
Comment (kadler) [inline CMakeLists.txt]: I'm guessing this should go down after OpenBSD to be in sorted order.

Comment (ThePrez) [reply CMakeLists.txt]: Thanks for noticing the alphabetization! Slid it down.

File: CMakeLists.txt [hunk 2]
```diff
@@ -242,6 +242,21 @@ if(CMAKE_SYSTEM_NAME STREQUAL "OS/390")
...
+       _LINUX_SOURCE_COMPAT
+       _THREAD_SAFE
+       _XOPEN_SOURCE=500)
+  list(APPEND uv_sources
+    src/unix/aix-common.c
+    src/unix/ibmi.c
+    src/unix/posix-poll.c
+    src/unix/no-fsevents.c
+    src/unix/no-proctitle.c
+    )
```
Comment (bnoordhuis) [inline CMakeLists.txt]:
```suggestion
    src/unix/no-fsevents.c
    src/unix/no-proctitle.c
    src/unix/posix-poll.c)
```

**Comment by richardlau** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1803/

**Comment by richardlau** [comment]
Landed in https://github.com/libuv/libuv/commit/104255f5af57153a9c590a002f53cfb6c15d4dd0.

**Pull request closed without merging**